perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,838 @@
+                               'second subnet for the DB instance.')
+
+    # Now create a new subnet in the zone that's different from where the VM is
+    logging.info('Creating a second subnet in zone %s', new_subnet_zone)
+    new_subnet = aws_network.AwsSubnet(new_subnet_zone, vm.network.vpc.id,
+                                       '10.0.1.0/24')
+    new_subnet.Create()
+    logging.info('Successfully created a new subnet, subnet id is: %s',
+                 new_subnet.id)
+    # Remember this so we can cleanup properly.
+    vm.extra_subnet_for_db = new_subnet
+
+    # Now we can create a new DB subnet group that has two subnets in it.
+    db_subnet_group_name = 'pkb%s' % FLAGS.run_uri
+    create_db_subnet_group_cmd = util.AWS_PREFIX + [
+        'rds',
+        'create-db-subnet-group',
+        '--db-subnet-group-name', db_subnet_group_name,
+        '--db-subnet-group-description', 'pkb_subnet_group_for_db',
+        '--subnet-ids', vm.network.subnet.id, new_subnet.id]
+    stdout, stderr, _ = vm_util.IssueCommand(create_db_subnet_group_cmd)
+    logging.info('Created a DB subnet group, stdout is:\n%s\nstderr is:\n%s',
+                 stdout, stderr)
+    vm.db_subnet_group_name = db_subnet_group_name
+
+    # open up tcp port 3306 in the VPC's security group, we need that to connect
+    # to the DB.
+    open_port_cmd = util.AWS_PREFIX + [
+        'ec2',
+        'authorize-security-group-ingress',
+        '--group-id', vm.group_id,
+        '--source-group', vm.group_id,
+        '--protocol', 'tcp',
+        '--port', '3306']
+    stdout, stderr, _ = vm_util.IssueCommand(open_port_cmd)
+    logging.info('Granted tcp 3306 ingress, stdout is:\n%s\nstderr is:\n%s',
+                 stdout, stderr)
+
+    # Finally, it's time to create the DB instance!
+    vm.db_instance_id = 'pkb-DB-%s' % FLAGS.run_uri
+    db_class = \
+        RDS_CORE_TO_DB_CLASS_MAP['%s' % FLAGS.mysql_svc_db_instance_cores]
+    vm.db_instance_master_user = MYSQL_ROOT_USER
+    vm.db_instance_master_password = _GenerateRandomPassword()
+
+    create_db_cmd = util.AWS_PREFIX + [
+        'rds',
+        'create-db-instance',
+        '--db-instance-identifier', vm.db_instance_id,
+        '--db-instance-class', db_class,
+        '--engine', RDS_DB_ENGINE,
+        '--engine-version', RDS_DB_ENGINE_VERSION,
+        '--storage-type', RDS_DB_STORAGE_TYPE_GP2,
+        '--allocated-storage', RDS_DB_STORAGE_GP2_SIZE,
+        '--vpc-security-group-ids', vm.group_id,
+        '--master-username', vm.db_instance_master_user,
+        '--master-user-password', vm.db_instance_master_password,
+        '--availability-zone', vm.zone,
+        '--db-subnet-group-name', vm.db_subnet_group_name]
+
+    status_query_cmd = util.AWS_PREFIX + [
+        'rds',
+        'describe-db-instances',
+        '--db-instance-id', vm.db_instance_id]
+
+    stdout, stderr, _ = vm_util.IssueCommand(create_db_cmd)
```

Comment:
The reason I don't use the return code is because AWS CLI does not document or guarantee any sort of return code, checking the return code is futile here and error-prone (e.g., they could return something non-zero to indicate status_pending but we'd thought it failed). What they do document is the stdout (the output of a command). e.g., for the create-db-instance command, they formally documented the output here: http://docs.aws.amazon.com/AmazonRDS/latest/CommandLineReference/CLIReference-cmd-CreateDBInstance.html

So I'm much more inclined to check the stdout (output) and validate the output is within expectation to judge success or failure, instead of relying on the return code.